src/build.rs

```diff
@@ -263,3 +265,323 @@ fn build_nasm(root: &Path, vendor_dir: &Path, out_dir: &Path, target_arch: &str,
     }
     n.compile_objects()
 }
+
+fn generate_ffi_mod() {
+    let out_dir = PathBuf::from(env::var_os("OUT_DIR").expect("outdir"));
+    let src_dir = env::current_dir().expect("failed to get current directory");
+
+    eprintln!("OUT_DIR={}", out_dir.display());
+    eprintln!("SRC_DIR={}", src_dir.display());
+    let ffi_input = src_dir.join("src").join("ffi.rs");
+    let ffi_src = std::fs::read_to_string(ffi_input.clone()).expect("failed to read src/ffi.rs");
+    println!("{}", format!("cargo:rerun-if-changed={}", ffi_input.display()));
+
+    // Parse src/ffi.rs generating Rust and C wrappers
+    let decls = parse_decls(&ffi_src);
```

Comment:
This parses the `ffi.rs` file, containing all the `extern "C" { ... }` declarations. You don't have to put them in a different file if you use, e.g., syn, to parse the `lib.rs` file, but if you just want to hack in your own parser, then doing this makes things simpler. I'd recommend just using `syn` here.